src/transactions/TransactionFrame.cpp

```diff
@@ -675,6 +675,18 @@ TransactionFrame::apply(Application& app, AbstractLedgerTxn& ltx,
         if (cv >= ValidationType::kInvalidUpdateSeqNum)
         {
             processSeqNum(ltxTx);
+
+            if (ltxTx.loadHeader().current().ledgerVersion >= 13 &&
+                cv == ValidationType::kInvalidUpdateSeqNum)
+            {
+                // check the operation signatures as well so we know which
+                // pre-auth signers to remove
+                for (auto& op : mOperations)
+                {
+                    op->checkSignature(signatureChecker, ltxTx, false, false);
```

Comment:
This seems fairly wasteful from a performance perspective, because we already know the transaction is invalid so checking signatures is unnecessary. The actual requirement is just to identify the used one-time signers so we can remove them. I wonder if it might be possible to expose this functionality by refactoring, allowing us to bypass `OperationFrame::checkSignature` here.